cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/AbstractContext.java

```diff
@@ -169,6 +172,27 @@
   protected final DynamicDatasetCache datasetCache;
   protected final RetryStrategy retryStrategy;
 
+
+  /**
+   * Returns runtime arguments as a Map.
+   * <ol>
+   *  <li>If FEATURE_FLAGS_IN_RUNTIME is set: extracting any feature flag defined in CConfiguration
+   *  Adding the userArguments from programOptions </li>
+   * </ol>
+   *
+   * @param programOptions
+   * @return a map of runtime arguments.
+   */
+  private Map<String, String> getRuntimeArgs(ProgramOptions programOptions) {
+    Map<String, String> runtimeArgs = new HashMap<>();
+    Map<String, String> conf = CConfigurationUtil.asMap(cConf);
+    if (Feature.FEATURE_FLAGS_IN_RUNTIME.isEnabled(conf)) {
+      runtimeArgs.putAll(FeatureFlagsUtil.extractFeatureFlags(conf));
```

Comment:
I don't think we should blend feature configs into the runtime arguments. We should have all program context extends from FeatureFlagProvider instead. A feature flags can be overridden by an argument from the runtime arguments.